### ISSUE TITLE
Cannot find objcopy

### ISSUE THREAD

**Issue by chenzepeng140** [ClickHouse/ClickHouse#47994]
> Make sure that `git diff` result is empty and you've just pulled fresh master. Try cleaning up cmake cache. Just in case, official build instructions are published here: https://clickhouse.com/docs/en/development/build/

**Operating system**
macOS 12.2.1

**Cmake version**
3.26.1
**Ninja version**
1.10.2
**Compiler name and version**
16.0.0
**Full cmake and/or ninja output**

(base) zepengchen@laojiekedeMacBook-Pro build % cmake -DCMAKE_C_COMPILER=$(brew --prefix llvm)/bin/clang -DCMAKE_CXX_COMPILER=$(brew --prefix llvm)/bin/clang++ -DCMAKE_BUILD_TYPE=RelWithDebInfo ..

-- The C compiler identification is Clang 16.0.0
-- The CXX compiler identification is Clang 16.0.0
-- The ASM compiler identification is Clang with GNU-like command-line
-- Found assembler: /opt/homebrew/opt/llvm/bin/clang
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /opt/homebrew/opt/llvm/bin/clang - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /opt/homebrew/opt/llvm/bin/clang++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Using compiler:
Homebrew clang version 16.0.0
Target: arm64-apple-darwin21.3.0
Thread model: posix
InstalledDir: /opt/homebrew/opt/llvm/bin
-- Using linker: <default>
-- Using archiver: /usr/bin/ar
-- Using ranlib: /opt/homebrew/opt/llvm/bin/llvm-ranlib
-- Using install-name-tool: /usr/bin/install_name_tool
CMake Error at cmake/tools.cmake:165 (message):
  Cannot find objcopy.
Call Stack (most recent call first):
  CMakeLists.txt:18 (include)

-- Configuring incomplete, errors occurred!

thx alot guys

**Comment by chenzepeng140** [comment]
(base) zepengchen@laojiekedeMacBook-Pro build % cmake -DCMAKE_C_COMPILER=$(brew --prefix llvm)/bin/clang -DCMAKE_CXX_COMPILER=$(brew --prefix llvm)/bin/clang++ -DCMAKE_BUILD_TYPE=RelWithDebInfo ..
-- Using compiler:
Homebrew clang version 16.0.0
Target: arm64-apple-darwin21.3.0
Thread model: posix
InstalledDir: /opt/homebrew/opt/llvm/bin
-- Using linker: <default>
-- Using archiver: /usr/bin/ar
-- Using ranlib: /opt/homebrew/opt/llvm/bin/llvm-ranlib
-- Using install-name-tool: /usr/bin/install_name_tool
CMake Error at cmake/tools.cmake:165 (message):
  Cannot find objcopy.
Call Stack (most recent call first):
  CMakeLists.txt:18 (include)


-- Configuring incomplete, errors occurred!

**Comment by chenzepeng140** [comment]
(base) zepengchen@laojiekedeMacBook-Pro build-aarch64 % cmake -DCMAKE_C_COMPILER=$(brew --prefix llvm)/bin/clang -DCMAKE_CXX_COMPILER=$(brew --prefix llvm)/bin/clang++ -DCMAKE_BUILD_TYPE=RelWithDebInfo ..
-- The C compiler identification is Clang 16.0.0
-- The CXX compiler identification is Clang 16.0.0
-- The ASM compiler identification is Clang with GNU-like command-line
-- Found assembler: /opt/homebrew/opt/llvm/bin/clang
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /opt/homebrew/opt/llvm/bin/clang - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /opt/homebrew/opt/llvm/bin/clang++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Using compiler:
Homebrew clang version 16.0.0
Target: arm64-apple-darwin21.3.0
Thread model: posix
InstalledDir: /opt/homebrew/opt/llvm/bin
-- Using linker: <default>
-- Using archiver: /usr/bin/ar
-- Using ranlib: /opt/homebrew/opt/llvm/bin/llvm-ranlib
-- Using install-name-tool: /usr/bin/install_name_tool
CMake Error at cmake/tools.cmake:165 (message):
  Cannot find objcopy.
Call Stack (most recent call first):
  CMakeLists.txt:18 (include)


-- Configuring incomplete, errors occurred!

**Comment by cangyin** [comment]
you should install objcopy.

or build inside the docker image `clickhouse/binary-builder`, which ships with all build tools.

**Comment by chenzepeng140** [comment]
> you should install objcopy.
> 
> or build inside the docker image `clickhouse/binary-builder`, which ships with all build tools.

thx

**Comment by rschu1ze** [comment]
> you should install objcopy.
> 
> or build inside the docker image `clickhouse/binary-builder`, which ships with all build tools.

Agree. For reference, the build searches for the `objcopy` binary here: https://github.com/ClickHouse/ClickHouse/blob/5d5308e060bcfe1d7a07cbebd2cc3eae833afe2d/cmake/tools.cmake#L157
